src/sidebar/components/menu-item.js

```diff
@@ -55,23 +55,33 @@ export default function MenuItem({
   }
   const leftIcon = isSubmenuItem ? null : renderedIcon;
   const rightIcon = isSubmenuItem ? renderedIcon : null;
+  // Only add `aria-checked` attribute if `isSubmenuItem` is false. Ensure that the value
+  // is a "true" or "false" string
+  const ariaChecked = !isSubmenuItem
+    ? { 'aria-checked': (!!isSelected).toString() }
+    : {};
 
   return (
     <Fragment>
-      {/* FIXME-A11Y */}
-      {/* eslint-disable-next-line jsx-a11y/role-supports-aria-props */}
       <div
-        aria-checked={isSelected}
         className={classnames('menu-item', {
           'menu-item--submenu': isSubmenuItem,
           'is-disabled': isDisabled,
           'is-expanded': isExpanded,
           'is-selected': isSelected,
         })}
         role="menuitem"
-        {...(onClick && onActivate('menuitem', onClick))}
       >
-        <div className="menu-item__action">
+        <div
+          className="menu-item__action"
```

Comment:
Basically I shifted the action stuff from the parent <div> to this <div> and this helps square away the aria attributes and role.